### ISSUE TITLE
store hash of file to tpm using cit

### ISSUE THREAD

**Issue by aforalee** [opencit/opencit#27]
According to my use case I need to get hash of some file to store it inside tpm using CIT, but if someone changes the file i want that trust status must changes to untrusted. So is this usecase possible to do using version 2.2, if yes Can you guys please guide me a little.

Thanks

**Comment by ryansavino** [comment]
This use case is not available in CIT 2.2, but is coming in CIT 3.2.

I'm assuming you're talking about the vm/container? Or are you referring to the physical host?

**Comment by aforalee** [comment]
Ryan thanks for your reply. 
Right now i am working with ubuntu 14.04 physical host.
but in future i also need this in vm and containers.
so is there any thing which can help.

Rightnow i am adding MLEPCRS values using Apis defined in java_docs.

 https://server.com:8181/mtwilson/v2/mles/31021a8a-de64-4c5f-b314-8d3f077a55e5/pcrs
 Input: {"pcr_index":"12","pcr_value":"9f3b7ae57724d6ef68ee6df89b4bfe002b042bfc"}


This is does update value in tpm.
i can see 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 by running
cat /sys/class/misc/tpm0/device/pcrs

**Comment by ryansavino** [comment]
400?

Why are you using port 8181? Can you try with port 8443? That's the default tomcat port, unless you changed it.

**Comment by ryansavino** [comment]
BTW, not sure what your use case is, but the javadoc quickstart page describes the simplified main functionality. You can find it here after you build (unzip, and you will see quickstart page):
integration/mtwilson-client-java7/target/mtwilson-client-java7-2.2-SNAPSHOT-javadoc.zip

Setup:
client registration > access approval > configure TLS policy

Functionality:
whitelist > host registration > trust verification

The quickstart describes the API client usage, but if you're building your own rest client, you can cross correlate the API client calls inside of the javadoc to see the REST implementations.

**Comment by aforalee** [comment]
Thanks 
i am using 8443 port. that was just an example.

Simple my usecase is to check integrity of file using cit.

I also have intel cit 3.0. According to my understanding i think i can fulfill this scenario using trust director policy. isn't it?

Thanks for your help

**Comment by ryansavino** [comment]
Yes, you would set the policy for file measurement of host and vm/container in the trust director application in CIT 3.x.

**Comment by aforalee** [comment]
i used CIT 3.0 it solved the issue but reboot is required if i change the file, so is there anything that i can do to avoid rebooting.
and is this rebooting thing is also needed in CIT 3.1 
I need to avoid rebooting and want that if file is changed i can refresh trust status from mtwilson and see machine as untrusted,

Thanks for your kind help.

**Comment by ryansavino** [comment]
Due to the nature of our product, runtime file attestation is not possible at this time. Can you work with your account rep to answer 3.x questions for time being? Thanks.

**Comment by aforalee** [comment]
Yeah sure. Thanks
